config.go

```diff
@@ -64,4 +65,17 @@ type Config struct {
 	// OnError is called if there's an issue with retrieving
 	// the response from the remote server.
 	OnError func(session *Session, err error)
+
+	// OnUnknownStream is called when we met unmitmable stream
+	OnUnknownStream func(session *Session, origin net.Conn) error
+
+	OnCheckMITM func(host string) bool
+}
+
+type Option func(p *Proxy)
+
+func WithTimeout(t time.Duration) Option {
```

Comment:
I don't really like the idea of providing a way to change the proxy configuration after it has been initialized. This will eventually lead to races.

Reply:
WithTimeout is only called when we need to init proxy via NewProxy, it is a init param and will not be used after init.

like this
```golang 
	proxy := gomitmproxy.NewProxy(gomitmproxy.Config{
		MITMConfig:      c,
		OnConnect:       s.onProxyConnect,
		OnRequest:       s.onProxyRequest,
		OnResponse:      s.onProxyResponse,
		OnError:         s.onProxyError,
		OnUnknownStream: s.onProxyUnknowStream,
		OnCheckMITM:     s.onCheckMITM,
	}, gomitmproxy.WithTimeout(0), gomitmproxy.WithXXXX(...), ...)
```

Reply:
It is also a common practice to use Option for initialization, which can avoid changing to old code, such as the `gomitmproxy.Config` you said. Of course, if you insist, I will also handle it according to your requirements.